lib/evmone/baseline.cpp

```diff
@@ -67,20 +68,36 @@ CodeAnalysis analyze_legacy(bytes_view code)
     return {pad_code(code), code.size(), analyze_jumpdests(code)};
 }
 
-CodeAnalysis analyze_eof1(bytes_view eof_container, const EOF1Header& header)
+CodeAnalysis analyze_eof1(bytes_view container)
 {
-    const auto executable_code = eof_container.substr(header.code_begin(), header.code_size);
-    return {executable_code, analyze_jumpdests(executable_code)};
+    const auto header = read_valid_eof1_header(container);
+
+    // Extract all code sections as single buffer reference.
+    // TODO: It would be much easier if header had code_sections_offset and data_section_offset
+    //       with code_offsets[] being relative to code_sections_offset.
+    const auto code_sections_offset = header.code_offsets[0];
+    const auto code_sections_end = size_t{header.code_offsets.back()} + header.code_sizes.back();
+    const auto executable_code =
+        container.substr(code_sections_offset, code_sections_end - code_sections_offset);
+
+    // Offsets relative to first code section.
```

Comment:
No. I think they have to be relative offsets to the beginning of the code sections in the container. This calculation is required because CodeAnalysis works on code sections as a single buffer reference. I added better comment. @chfast has also some comments here. There is also proper TODO added to this function.